.github/workflows/chainloop_github_release.yml

```diff
@@ -0,0 +1,116 @@
+# Description: This workflow is used to record a release in Chainloop. It is triggered when a release is published on GitHub.
+# It uses the Chainloop CLI to create an attestation for the release assets, source code and additional materials if provided.
+# The attestation is then pushed to the Chainloop service and the attestation link is added to the release notes.
+# Prior to running this workflow, the Chainloop workflow is onboarded if it does not exist.
+on:
+  workflow_call:
+    inputs:
+      project:
+        description: "The project this workflow belongs to. default: repository name"
+        type: string
+      workflow_name:
+        description: "The workflow name. default: parent workflow filename"
+        type: string
+      additional_materials:
+        description: "Additional materials to be attested. Commas separated list of values"
+        type: string
+    secrets:
+      api_token:
+        description: "Reference: https://docs.chainloop.dev/reference/operator/api-tokens#api-tokens"
+        required: true
+      cosign_key:
+        description: "The private key used to sign the attestation"
+        required: true
+      cosign_password:
+        description: "The password for the private key used to sign the attestation"
+        required: true
+
+jobs:
+  onboard_workflow:
+    name: Onboard Chainloop Workflow
+    uses: chainloop-dev/labs/.github/workflows/chainloop_onboard.yml@4173e015dbd5dc2a8802555c268da63d57bbe576
+    if: github.event_name == 'release' && github.event.action == 'published'
+    with:
+      project: ${{ inputs.workflow_project }}
+      workflow_name: ${{ inputs.workflow_name }}
+    # Pass parent workflow secrets to the child workflow
+    secrets: inherit
+
+  release:
+    name: Record release from GitHub
+    runs-on: ubuntu-latest
+    needs: onboard_workflow
+    # Only run this job when a release is published
+    if: github.event_name == 'release' && github.event.action == 'published'
+    env:
+      CHAINLOOP_WORKFLOW_NAME: ${{ inputs.workflow_name }}
```

Comment:
would this work if inputs.workflow_name is not set? 

Have you tried to run this workflow with just the APIToken + Cosign keys? Does it work?

Reply:
It's actually `needs.onboard_workflow.outputs.workflow_name` not the one there, I'll change it.

With that change it will take whatever was passed to the onboarding workflow that by default if not set, sets a `workflow_name`

Reply:
It works but we have this bug: https://github.com/chainloop-dev/labs/issues/19